src/mame/drivers/brother_ax145.cpp

```diff
@@ -0,0 +1,352 @@
+		map(0x4000, 0x4000).r(FUNC(ax145_state::io_4000_r)); // config upper 4 bits
+		map(0x5000, 0x5000).w(FUNC(ax145_state::io_5000_w)); // dictionary banking
+		map(0x5800, 0x5800).w(FUNC(ax145_state::lcd_signal_w));
+		map(0x6000, 0x6000).rw(FUNC(ax145_state::lcd_data_r), FUNC(ax145_state::lcd_data_w));
+		//map(0x6800, 0x6800).w(TODO);
+		//map(0x7000, 0x7000).w(TODO);
+		//map(0x7800, 0x7800).w(TODO);
+		//map(0x8000, 0x8000).w(TODO);
+		//map(0x8800, 0x8800).w(TODO);
+		//map(0x9000, 0x9000).r(TODO);
+		map(0xb000, 0xb000).r(FUNC(ax145_state::irq_ack_r));
+		//map(0xb800, 0xb800).rw(TODO, TODO);
+	}
+
+	void palette(palette_device &palette) const {
+		palette.set_pen_color(0, rgb_t(138, 146, 148));
+		palette.set_pen_color(1, rgb_t(92, 83, 88));
+	}
+
+	uint8_t* rom{};
+	uint8_t* dict_rom{};
+	std::map<uint32_t, std::string> symbols;
+};
+
+void ax145_state::video_start()
+{
+}
+
+std::string ax145_state::pc()
+{
+	class z180_friend : public z180_device { public: using z180_device::memory_translate; friend class ax145_state; };
+	auto cpu = static_cast<z180_friend*>(dynamic_cast<z180_device*>(&machine().scheduler().currently_executing()->device()));
+	offs_t phys = cpu->pc();
+	cpu->memory_translate(AS_PROGRAM, 0, phys);
+
+	return symbolize(phys);
+}
+
+std::string ax145_state::symbolize(uint32_t adr)
+{
+	if(symbols.empty())
+		return string_format("%06x", adr);
+
+	auto floor_it = symbols.lower_bound(adr);
+	if((floor_it == symbols.end() && !symbols.empty()) || floor_it->first != adr)
+		--floor_it;
+	if(floor_it != symbols.end())
+		return string_format("%s+%x (%06x)", floor_it->second, adr - floor_it->first, adr);
+	else
+		return string_format("%06x", adr);
+}
+
+std::string ax145_state::callstack()
+{
+	class z180_friend : public z180_device { public: using z180_device::memory_translate; friend class ax145_state; };
+	auto cpu = static_cast<z180_friend*>(dynamic_cast<z180_device*>(&machine().scheduler().currently_executing()->device()));
+	offs_t pc = cpu->pc();
+	cpu->memory_translate(AS_PROGRAM, 0, pc);
+
+	//int depth = 0;
+	std::string output;
+	output += symbolize(pc) + " >> ";
+
+//	if(output.find("abort") != std::string::npos)
+//		__debugbreak();
+
+	return output;
+}
+
+void ax145_state::machine_start()
+{
+/*	// try to load map file
+	FILE* f;
+	if(fopen_s(&f, "ax145.map", "rt") == 0) {
+		char line[512];
+		do {
+			if(fgets(line, sizeof(line), f)) {
+				int segment, offset;
+				char symbol[512];
+				if(sscanf(line, "%x:%x %512s", &segment, &offset, symbol) == 3) {
+					uint32_t phys = (segment << 4) + offset;
+					//TRACE(_T("%04x:%04x => %02x:%04x\n"), segment, offset, bank, offset);
+					symbols[phys] = symbol;
+				}
+			}
+		} while(!feof(f));
+		fclose(f);
+	}
+*/
+
+	maincpu->space(AS_PROGRAM).install_ram(0x02000, 0x03fff, ram); // first 0x2000 bytes of RAM
+	maincpu->space(AS_PROGRAM).install_ram(0x62000, 0x69fff, ram); // complete 0x8000 bytes of RAM
```

Comment:
You should be able to do this with within the memory map using `.share`

Reply:
I tried that before, but then the share got implicitly created using 0x2000 bytes size, and when it encountered the second region, it complained about the size mismatch.
tried memory_share_creator but failed. Any idea?